x/authority/types/authorizations_test.go

```diff
@@ -0,0 +1,306 @@
+package types_test
+
+import (
+	"fmt"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+
+	"github.com/zeta-chain/zetacore/x/authority/types"
+	crosschaintypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+	fungibletypes "github.com/zeta-chain/zetacore/x/fungible/types"
+	lightclienttypes "github.com/zeta-chain/zetacore/x/lightclient/types"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+func TestAuthorizationList_SetAuthorizations(t *testing.T) {
+	tt := []struct {
+		name             string
+		oldList          types.AuthorizationList
+		addAuthorization types.Authorization
+		expectedList     types.AuthorizationList
+	}{
+		{
+			name: "set new authorization successfully",
+			oldList: types.AuthorizationList{Authorizations: []types.Authorization{
+				{
+					MsgUrl:           "ABC",
+					AuthorizedPolicy: types.PolicyType_groupOperational,
+				},
+			}},
+			addAuthorization: types.Authorization{
+				MsgUrl:           "XYZ",
+				AuthorizedPolicy: types.PolicyType_groupOperational,
+			},
+			expectedList: types.AuthorizationList{Authorizations: []types.Authorization{
+				{
+					MsgUrl:           "ABC",
+					AuthorizedPolicy: types.PolicyType_groupOperational,
+				},
+				{
+					MsgUrl:           "XYZ",
+					AuthorizedPolicy: types.PolicyType_groupOperational,
+				},
+			}},
+		},
+		{
+			name: "update existing authorization successfully",
+			oldList: types.AuthorizationList{Authorizations: []types.Authorization{
+				{
+					MsgUrl:           "ABC",
+					AuthorizedPolicy: types.PolicyType_groupOperational,
+				},
+			}},
+			addAuthorization: types.Authorization{
+				MsgUrl:           "ABC",
+				AuthorizedPolicy: types.PolicyType_groupEmergency,
+			},
+			expectedList: types.AuthorizationList{Authorizations: []types.Authorization{
+				{
+					MsgUrl:           "ABC",
+					AuthorizedPolicy: types.PolicyType_groupEmergency,
+				},
+			}},
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.oldList.SetAuthorizations(tc.addAuthorization)
+			require.Equal(t, tc.expectedList, tc.oldList)
+		})
+	}
+}
+
+func TestAuthorizationList_GetAuthorizations(t *testing.T) {
+	tt := []struct {
+		name            string
+		authorizations  types.AuthorizationList
+		getPolicyMsgUrl string
+		expectedPolicy  types.PolicyType
+		error           error
+	}{
+		{
+			name: "get authorizations successfully",
```

Comment:
Should add tests retrieve a admin and emergency policy
Add a test where we search for an element in a non empty list